Adding github actions file to build and push docker image to docker hub / ECR

adityabharadwaj198 opened the pull request:
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
*  Refactoring


* **What is the current behavior?** (You can also link to an open issue here)
Currently the same github action exists in the open_source_CI repo (https://github.com/marqo-ai/open_source_CI/blob/main/.github/workflows/build_push_img.yml). We want to move it here.

* **What is the new behavior (if this is a feature change)?**
Behavior remains the same - the action can now be invoked from main Marqo repo

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
No

* **Related Python client changes** (link commit/PR here)
NA

* **Related documentation changes** (link commit/PR here)
NA

* **Other information**:


* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

adityabharadwaj198 committed:
Adding github actions file to build and push docker image to docker hub / ECR

wanliAlex commented on .github/workflows/build_push_img.yml:
```diff
@@ -0,0 +1,132 @@
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_EC2_GH_RUNNER_ACCESS_KEY_ID }}
```
What is the AWS account? OpenSource or S2Search Core?

adityabharadwaj198 replied:
This is open source now - I changed the secrets to use opensource

wanliAlex commented on .github/workflows/build_push_img.yml:
```diff
@@ -0,0 +1,132 @@
+        required: false
+        default: 'mainline'
+      push_to:
+        description: 'image destination. Options: "ECR", "DockerHub"'
```
Make them drop down options?

adityabharadwaj198 replied:
Good suggestion, Sure!

adityabharadwaj198 replied:
done

adityabharadwaj198 committed:
Made options available as drop down & changed secrets to use new secrets defined in PR: https://github.com/marqo-ai/marqo/pull/997/files

The secrets are for GhRunnerManager-Marqo in open source account which has access to start new instances based on the AMI provided. The AMI also exists in the open source account

wanliAlex commented on .github/workflows/build_push_img.yml:
```diff
@@ -0,0 +1,135 @@
+        with:
+          aws-access-key-id: ${{ secrets.MARQO_WORKFLOW_TESTS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.MARQO_WORKFLOW_TESTS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
```
Why do we start the instance on us-west-2 here? Should it also be on us-east-1?

wanliAlex commented on .github/workflows/build_push_img.yml:
```diff
@@ -0,0 +1,135 @@
+          ec2-image-id:  ami-01accacd51bdde263
+          ec2-instance-type: t3.xlarge
+          subnet-id: subnet-038b1447ac3c97e7a
+          security-group-id: sg-094be521399e0d5ba
```
Maybe add some tags like this? https://github.com/marqo-ai/marqo/blob/5c372ddc48498a50ea1fb8f53364644f6fbefcaa/.github/workflows/unit_test_200gb_CI.yml#L47-L55

adityabharadwaj198 replied:
Good idea, done.

adityabharadwaj198 committed:
Adding aws-resource-tags

adityabharadwaj198 committed:
Merge branch 'mainline' into aditya/github-actions-build-push-image
